parsing.js

```diff
@@ -222,11 +244,63 @@ function findIntroStart(content) {
  *
  * @param {Object} document
  * @param {int} verse_loc
- * @returns
+ * @returns text array wich contains all the text in between start and end of linguistic meaning section
  */
 function parseLinguistics(document, verse_loc) {
-    let lings = {};
-    return lings;
+    let content = document.body.content;
+
+    // get the current start and end of the Linguistic Meaning section
+    // based on verse_loc index
+    let startIndex = getVerseSectionStart(verse_loc, content);
+    let endIndex = getVerseSectionEnd(startIndex, content);
+
+    // return the text between the start and end indices (JSON)
+    return getTextInBetween(startIndex, endIndex, content);
+}
+
+/**
+ * A function that focuses on finding the beginning of a verse section
+ *
+ * @param {Object} content
+ * @returns line_index, which is the index where the current verse section begins
+ */
+function getVerseSectionStart(verse_loc, content) {
```

Comment:
Instead of having a separate method that does mostly the same thing, follow the DRY principle and see if you can combine both start and end methods into one.

Reply:
Will do